src/config.c

```diff
@@ -3068,7 +3068,7 @@ standardConfig static_configs[] = {
     createBoolConfig("aof-disable-auto-gc", NULL, MODIFIABLE_CONFIG | HIDDEN_CONFIG, server.aof_disable_auto_gc, 0, NULL, updateAofAutoGCEnabled),
     createBoolConfig("replica-ignore-disk-write-errors", NULL, MODIFIABLE_CONFIG, server.repl_ignore_disk_write_error, 0, NULL, NULL),
     createBoolConfig("extended-redis-compatibility", NULL, MODIFIABLE_CONFIG, server.extended_redis_compat, 0, NULL, updateExtendedRedisCompat),
-    createBoolConfig("enable-debug-assert", NULL, IMMUTABLE_CONFIG | HIDDEN_CONFIG, server.enable_debug_assert, 0, NULL, NULL),
+    createBoolConfig("debug-assert-enabled", NULL, IMMUTABLE_CONFIG | HIDDEN_CONFIG, server.debug_assert_enabled, 0, NULL, NULL),
```

Comment:
What exactly is the convention here? We have other "debug" like configs like `rdb-key-save-delay` (this is used to artificially slow down snapshots for tests). What do we consider a "debug" config?

Reply:
>What do we consider a "debug" config?

Anything not meant to be used in the production. I haven't gone through all the debug settings though.

Reply:
Maybe let's start with all the "HIDDEN_CONFIG" then? That was at least my thinking when we added that, it would be a proxy for the ones not intended for production.

Reply:
Make sense. Added

Reply:
Can you elaborate the unnecessary work part? It is always a judgement call when it comes to improving the code quality. I don't want to go with a blanket statement of either refactoring everything or nothing should be changed. "Incremental perfection" that "leaves the ground cleaner than found" is my principle.

Reply:
There are additional test suites where this config is used and needs to be updated. Could we maybe alias it, so that it's not a breaking change.

Reply:
I believe we are keeping the config names as is. The only one that is proposed to get changed is the debug assert one, which hasn't been released yet.